pennylane/devices/default_tensor.py

```diff
@@ -810,14 +812,24 @@ def _local_expectation(self, matrix, wires) -> float:
         # after the execution, we could avoid copying the circuit.
         qc = self._quimb_circuit.copy()
 
-        exp_val = qc.local_expectation(
-            matrix,
-            wires,
-            dtype=self._c_dtype.__name__,
-            optimize=self._contraction_optimizer,
-            simplify_sequence=self._local_simplify,
-            simplify_atol=0.0,
-        )
+        if self.method == "mps":
+            exp_val = qc.local_expectation(
+                matrix,
+                wires,
+                dtype=self._c_dtype.__name__,
+                optimize=self._contraction_optimizer,
+            )
+        elif self.method == "tn":
+            exp_val = qc.local_expectation(
+                matrix,
+                wires,
+                dtype=self._c_dtype.__name__,
+                optimize=self._contraction_optimizer,
+                simplify_sequence=self._local_simplify,
+                simplify_atol=0.0,
+            )
+        else:
```

Comment:
Note that it is logically impossible to reach this line, hence the `# pragma: no cover`. It could also be removed completely with a simple `if-else`

Reply:
`DefaultTensor.__init__` already validates that the method is either `mps` or `tn`, so we don't need this `else` branch.

Reply:
Yes, it is totally useless (I actually don't know why I inserted it in the first place). Removed now